### PR TITLE
Add minimumLimaVersion and vmOpts.qemu.minimumVersion fields to lima.yaml

### PR DESCRIPTION
**Pull request by jandubois**
This will allow us to specify the requirements in the templates not just in a comment, but in a way that can be validated.

### CODE REVIEW

**Comment by jandubois** [comment]
I noticed we use inconsistent capitalization of the QEMU name, sometimes even in the same type:

```
func (l *LimaQemuDriver) killQEMU(…)
```

I feel that completely capitalizing it as `QEMU` is the right way to use it, but we should do this consistently. If people prefer `Qemu`, then I can use this in the new variables too, but we should then rename all the old names to match as well.

File: examples/default.yaml [hunk 1]
```diff
@@ -5,6 +5,16 @@
...
+# A template should specify the minimum Lima version required to parse this template correctly.
+# It should not be set if the minimum version is less than 1.0.0
+# 🟢 Builtin default: not set
+minimumLimaVersion: null
+
+# Minimum version of QEMU required to create an instance of this template.
+# Will be ignored if the vmType is not "qemu"
+# 🟢 Builtin default: not set
+minimumQEMUVersion: null
+
```
Comment (AkihiroSuda) [inline examples/default.yaml:17]:
```suggestion
vmOpts:
  qemu:
    minimumVersion: null
```

Comment (jandubois) [reply examples/default.yaml:17]: Feels a bit too deeply nested right now, but I guess we can put additional stuff (something like `QEMU_SYSTEM_X86_64`) under there as well in the future.

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**